rmf_traffic_ros2/include/rmf_traffic_ros2/schedule/Negotiation.hpp

```diff
@@ -68,6 +68,48 @@ class Negotiation
   /// Get the current timeout duration setting.
   rmf_traffic::Duration timeout_duration() const;
 
+  using TableViewPtr = rmf_traffic::schedule::Negotiation::Table::ViewerPtr;
+  using StatusUpdateCallback =
+    std::function<void (uint64_t conflict_version, TableViewPtr table_view)>;
+  
+  /// Register a callback with this Negotiation manager that triggers
+  /// on negotiation status updates.
+  ///
+  /// \param[in] cb
+  ///   The callback function to be called upon status updates.
+  void on_status_update(StatusUpdateCallback cb);
+
+  using StatusConclusionCallback =
+    std::function<void (uint64_t conflict_version, bool success)>;
+  
+  /// Register a callback with this Negotiation manager that triggers
+  /// on negotiation status conclusions.
+  ///
+  /// \param[in] cb
+  ///   The callback function to be called upon status conclusions.
+  void on_conclusion(StatusConclusionCallback cb);
+
+  /// Get a Negotiation::TableView that provides a view into what participants are
+  /// proposing.
+  ///
+  /// This function does not care about table versioning.
+  /// \param[in] conflict_version
+  ///   The conflict version of the negotiation
+  /// \param[in] sequence
+  ///   The sequence of participant ids. Follows convention of other sequences
+  ///   (ie. The last ParticipantId is the owner of the table)
+  ///
+  /// \return A TableView into what participants are proposing.
+  TableViewPtr table_view(
```

Comment:
Since this function shouldn't modify anything about the internal state, we should be able to make this a `const` member function.

Reply:
fixed a7c6ccb